documentation/guide-angular-mock-service.asciidoc

```diff
@@ -0,0 +1,84 @@
+:toc: macro
+
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+endif::[]
+
+toc::[]
+:idprefix:
+:idseparator: -
+:reproducible:
+:source-highlighter: rouge
+:listing-caption: Listing
+
+= Angular Mock Service
+We've all been there: A new idea comes, let's quickly prototype it. But wait, there's no backend. What can we do? 
+
+Well... we can write a simple mock service that helps us by feeding data into our components. Below, I will elaborate on the idea.
+
+== The app we start with
+Let's say you have a simple boilerplate code, with your favorite styling library hooked up and you're ready to go. I chose the https://github.com/devonfw/devon4ng/tree/develop/samples/AngularMaterialBasicLayout[Angular Material] sample to start with. 
+
+== The Components
+Components - are the building blocks of our application. Their main role is to enable fragments of user interfaces. They will eaither display data (a list , a table, a chart, etc.), or 'collect' user interaction (e.g: a form, a menu, etc.)
+Components stay at the forefront of the application. They should also be reusable (as much as possible). Reusability is key for what we are trying to achieve - a stable, maintainable frontend where multiple people can contribute and collaborate. 
+
+In our project, we are at the beginning. That means we may have more ideas than plans. We are exploring possibilites. In order to code eficiently: +
+1) We will not store mock data in the components. +
+2) We will not fetch or save data directly in the components.  
+
+https://angular.io/start#components[Learn more about Angular Components]
```

Comment:
Please refer to the current devon4ng documentation. For example here you can refer to https://github.com/devonfw/devon4ng/wiki/components-layer